src/Utilities/Utilities.jl

```diff
@@ -363,15 +363,34 @@ end
 
 # Finding URLs -- based partially on code from the main Julia repo in `base/methodshow.jl`.
 
+function in_cygwin()
+    # check if in cygwin, if so we need to translate cygwin paths to windows paths
+    if is_windows()
+        try
+            success(`cygpath -h`)
+        catch
+            return false
+        end
+    else
+        return false
+    end
+end
+
 function url(repo, file)
     file = abspath(file)
     remote = getremote(dirname(file))
     isempty(repo) && (repo = "https://github.com/$remote/tree/{commit}{path}")
     commit, root = cd(dirname(file)) do
-        readchomp(`git rev-parse HEAD`), readchomp(`git rev-parse --show-toplevel`)
+        toplevel = readchomp(`git rev-parse --show-toplevel`)
+        if in_cygwin()
```

Comment:
I am setting `JULIA_PKGDIR=$HOME/julia`. It seems that then Julia gets UNIX-y paths, but uses the Windows path separator to join them together, or something along those lines? Already the `Pkg.dir()` is weird:
```
$ JULIA_PKGDIR=$HOME/julia /cygdrive/c/Users/morten/julia/julia-bb76add105/bin/julia
julia> print(Pkg.dir())
\home\morten\julia\v0.6
```